test/plain.jl

```diff
@@ -250,6 +250,21 @@ g = fid["mygroup"]
 close(g)
 close(fid)
 
+# more do syntax: atomic rename version
+if VERSION >= v"0.4"
+    h5rewrite(fn) do fid
+        g_create(fid, "mygroup") do g
```

Comment:
This should test both important cases -- the case where the user function succeeds, and the case where it fails. It should probably also check whether the temporary file has been removed. (You can put everything into a directory, and then check how many files are in there.)